src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/UnconfiguredProjectContextProvider.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+using Microsoft.VisualStudio.LanguageServices.ProjectSystem;
+using Microsoft.VisualStudio.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices
+{
+    /// <summary>
+    ///     Implementation of <see cref="IProjectContextProvider"/> that creates an <see cref="IWorkspaceProjectContext"/>
+    ///     based on the an <see cref="UnconfiguredProject"/>.
+    /// </summary>
+    [Export(typeof(IProjectContextProvider))]
+    internal partial class UnconfiguedProjectContextProvider : OnceInitializedOnceDisposed, IProjectContextProvider
+    {
+        private readonly IUnconfiguredProjectCommonServices _commonServices;
+        private readonly Lazy<IWorkspaceProjectContextFactory> _contextFactory;
+        private readonly IProjectAsyncLoadDashboard _asyncLoadDashboard;
+        private readonly ITaskScheduler _taskScheduler;
+        private readonly List<IWorkspaceProjectContext> _contexts = new List<IWorkspaceProjectContext>();
```

Comment:
In that case, why are we changing the ProjectFilePath and DisplayName for all contexts in OnProjectRenamed? Shouldn't it be just for the single context corresponding to the project that was renamed?